File: src/main/java/com/mycompany/myapp/service/dto/CardDTO.java
package com.mycompany.myapp.service.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Created by derekzuk on 10/10/17.
 */
public class CardDTO {

    @JsonProperty("number")
    private Long number;

    @JsonProperty("expMonth")
    private Long expMonth;

    @JsonProperty("expYear")
    private Long expYear;

    @JsonProperty("cvc")
    private Long cvc;

    public Long getNumber() {
        return number;
    }

    public void setNumber(Long number) {
        this.number = number;
    }

    public Long getExpMonth() {
        return expMonth;
    }

    public void setExpMonth(Long expMonth) {
        this.expMonth = expMonth;
    }

    public Long getExpYear() {
        return expYear;
    }

    public void setExpYear(Long expYear) {
        this.expYear = expYear;
    }

    public Long getCvc() {
        return cvc;
    }

    public void setCvc(Long cvc) {
        this.cvc = cvc;
    }

    @Override
    public String toString() {
        return "CardDTO{" +
            "number=" + number +
            ", expMonth=" + expMonth +
            ", expYear=" + expYear +
            ", cvc=" + cvc +
            '}';
    }
}

File: src/main/webapp/app/shared/dto/card.dto.ts
export class CardDTO {
        public ccOwner?: string;
        public ccNumber?: string;
        public expMonth?: string;
        public expYear?: string;
        public cvc?: string;

    constructor(
        ccOwner?: string,
        ccNumber?: string,
        expMonth?: string,
        expYear?: string,
        cvc?: string
    ) {
        this.ccOwner = ccOwner ? ccOwner : null;
        this.ccNumber = ccNumber ? ccNumber : null;
        this.expMonth = expMonth ? expMonth : null;
        this.expYear = expYear ? expYear : null;
        this.cvc = cvc ? cvc : null;
    }
}
